datasets/za/pmg_legislators/crawler.py

```diff
@@ -0,0 +1,165 @@
+
+# List of regex patterns for positions of interest
+POSITIONS_OF_INTEREST = [
+    re.compile(r"National\s+Legislature", re.IGNORECASE),
+    re.compile(r"National\s+Assembly", re.IGNORECASE),
+    re.compile(r"National\s+Council of Provinces", re.IGNORECASE),
+    re.compile(r"Minister of", re.IGNORECASE),
+    re.compile(r"Provincial\s+Legislature", re.IGNORECASE),
+    re.compile(r"Member of the Provincial Legislature", re.IGNORECASE),
+    re.compile(r"Member of the Executive Committee", re.IGNORECASE),
+    re.compile(r"National\s+Executive", re.IGNORECASE),
+]
+
+
+def clean_emails(emails):
+    out = []
+    for email in h.multi_split(emails, ["/", ",", " or "]):
+        if email is None:
+            return
+        email = unquote(email)
+        email = email.strip()
+        email = email.rstrip(".")
+        out.append(email)
+    return out
+
+
+def clean_phones(phones):
+    out = []
+    for phone in h.multi_split(phones, PHONE_SPLITS):
+        phone = PHONE_REMOVE.sub("", phone)
+        out.append(phone)
+    return out
+
+
+def parse_person(context: Context, data: dict):
+    person_id = data.get("id")
+    if not person_id:
+        context.log.error(f"Missing ID in person data: {data}")
+        return
+    person = context.make("Person")
+    person.id = person_entity_id(context, person_id)
+    person.add("name", data.get("name"))
+    person.add("alias", [o.get("name") for o in data.get("other_names", [])])
+    person.add("gender", data.get("gender"))
+    person.add("firstName", data.get("given_name"))
+    person.add("lastName", data.get("family_name"))
+    person.add("birthDate", data.get("birth_date"))
+    person.add("deathDate", data.get("death_date"))
+    person.add("notes", data.pop("summary", None))
+    person.add("title", data.pop("honorific_prefix", None))
+    person.add("topics", "role.pep")
+
+    for link in data.pop("links", []):
+        url = link.get("url")
+        if link.get("note") in ("website", "blog", "twitter", "facebook"):
+            person.add("website", url)
+
+    for contact_detail in data.pop("contact_details", []):
+        value = contact_detail.get("value")
+        if "email" == contact_detail.get("type"):  # might want to remove later
+            person.add("email", clean_emails(value))
+        if "phone" == contact_detail.get("type"):
+            person.add("phone", clean_phones(value))
+        if "cell" == contact_detail.get("type"):
+            person.add("phone", clean_phones(value))
+        if "address" == contact_detail.get("type"):
+            person.add("address", value)
+        if "postal_address" == contact_detail.get("type"):
+            person.add("address", value)
+
+    for ident in data.pop("identifiers", []):
+        identifier = ident.get("identifier")
+        scheme = ident.get("scheme")
+        if scheme == "wikidata" and identifier.startswith("Q"):
+            person.add("wikidataId", identifier)
```

Comment:
nice! we also want to make the person.id their QID if rigour.ids.wikidata.qid(id) is true - you can use that instead of startswith here.